core/src/alpenglow_consensus/certificate_pool.rs

```diff
@@ -260,10 +260,17 @@ impl<VC: VoteCertificate> CertificatePool<VC> {
         slot: Slot,
         vote_type: VoteType,
         validator_vote_key: &Pubkey,
+        vote_key: Option<VoteKey>,
     ) -> Option<VoteType> {
         for conflicting_type in conflicting_types(vote_type) {
             if let Some(pool) = self.vote_pools.get(&(slot, *conflicting_type)) {
-                if pool.has_prev_vote(validator_vote_key) {
+                if pool.has_prev_vote(
+                    validator_vote_key,
+                    conflicting_type
+                        .is_block_type()
+                        .then_some(vote_key.as_ref())
+                        .flatten(),
```

Comment:
I believe only `Notarize` and `NotarizeFallback` have vote_key not equal to `(None, None)`, so it's also possible that you blindly pass down `vote_key` and then check whether `vote_key.bank_hash` is `None` inside `vote_pool.rs`, then you don't really need `.is_block_type()`, and `vote_key` can just be of type `VoteKey` instead of `Option<VoteKey>`.